shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -655,96 +668,227 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
+               current_composition_params = current_composition_params_,   //
+               views_to_recomposite = views_to_recomposite_,               //
+               callbacks = std::move(callbacks),                           //
+               composition_order = composition_order_,                     //
+               unused_layers = std::move(unused_layers),
+               views_to_dispose = DisposeViews()  //
+  ]() mutable {
+    PerformSubmit(platform_view_layers,        //
+                  callbacks,                   //
+                  current_composition_params,  //
+                  views_to_recomposite,        //
+                  composition_order,           //
+                  unused_layers,               //
+                  views_to_dispose             //
+    );
+  };
+
+  // Workaround for FlutterPlatformViewsTest.mm
+  if ([[NSThread currentThread] isMainThread]) {
+    task();
+  } else {
+    platform_task_runner_->PostTask(task);
+  }
+
+  return did_submit;
+}
+
+void FlutterPlatformViewsController::CreateMissingOverlays(
+    GrDirectContext* gr_context,
+    const std::shared_ptr<IOSContext>& ios_context,
+    size_t required_overlay_layers) {
+  TRACE_EVENT0("flutter", "FlutterPlatformViewsController::CreateMissingLayers");
+
+  auto missing_layer_count = required_overlay_layers - layer_pool_->size();
+  // Workaround for FLutterPlatformViewsTest
+  if ([[NSThread currentThread] isMainThread]) {
+    // Create Missing Layers
+    for (auto i = 0u; i < missing_layer_count; i++) {
+      CreateLayer(gr_context,                                      //
+                  ios_context,                                     //
+                  ((FlutterView*)flutter_view_.get()).pixelFormat  //
+      );
+    }
+    return;
+  }
+
+  auto latch = std::make_shared<fml::CountDownLatch>(1u);
+  platform_task_runner_->PostTask([&]() {
+    // Create Missing Layers
+    for (auto i = 0u; i < missing_layer_count; i++) {
+      CreateLayer(gr_context,                                      //
+                  ios_context,                                     //
+                  ((FlutterView*)flutter_view_.get()).pixelFormat  //
+      );
+    }
+    latch->CountDown();
+  });
+  latch->Wait();
+}
+
+/// Update the buffers and mutate the platform views in CATransaction on the platform thread.
+void FlutterPlatformViewsController::PerformSubmit(
+    const LayersMap& platform_view_layers,
+    const std::vector<SurfaceFrame::DeferredSubmit>& callbacks,
+    std::map<int64_t, EmbeddedViewParams>& current_composition_params,
+    const std::unordered_set<int64_t>& views_to_recomposite,
+    const std::vector<int64_t>& composition_order,
+    const std::vector<std::shared_ptr<FlutterPlatformViewLayer>>& unused_layers,
+    const std::vector<UIView*>& views_to_dispose) {
+  TRACE_EVENT0("flutter", "FlutterPlatformViewsController::SubmitFrame::CATransaction");
```

Comment:
Do we have a handle to the current task runner? Perhaps DCHECK for running on the platform view?